Session03/matplotlib.ipynb

```diff
@@ -0,0 +1,610 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Session 03: Intro to Matplotlib Plotting on the Science Platform\n",
+    "\n",
+    "<br>Owner(s): **Keith Bechtol** ([@bechtol](https://github.com/LSSTScienceCollaborations/StackClub/issues/new?body=@bechtol))\n",
+    "<br>Last Verified to Run: **2020-05-19**\n",
+    "<br>Verified Stack Release: **w_2020_17**\n",
+    "\n",
+    "This notebook is intended as a warm-up to the Visualization lesson (Lesson 3), providing a brief introduction to data visualization with [matplotlib](https://matplotlib.org/). Matplotlib is one of the most widely used libraries for data visualization in astrophysics, and extensive documentation as well as many [examples](https://matplotlib.org/gallery/index.html) can be found with a quick websearch. Matplotlib is also part of the [PyViz](https://pyviz.org/) suite of visualization tools for python. This notebook walks through a few examples to get you started quickly plotting data from Rubin Observatory and precursor datasets.\n",
+    "\n",
+    "Today we'll cover:\n",
+    "* How to create a few common types of plots for tabular data (histograms, scatter plots)\n",
+    "* How to customize plot style, e.g., colors, markerstyle, axis labels, legends, etc.\n",
+    "\n",
+    "We'll use the same datasets "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# What version of the Stack am I using?\n",
+    "! echo $HOSTNAME\n",
+    "! eups list -s lsst_distrib"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Preliminaries"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's begin by importing plotting packages. Right away, we are faced with a choice as to which [backend](https://matplotlib.org/faq/usage_faq.html#what-is-a-backend) to use for plotting. For this demo, we'll use the [ipympl](https://github.com/matplotlib/ipympl) backend that allows us to create interactive plots (e.g., pan, zoom, and resize canvas capability) in a JupyterLab notebook. This option is enabled with the line\n",
+    "\n",
+    "```%matplotlib widget```\n",
```

Comment:
`%matplotlib notebook` throws an error for me
`Javascript Error: IPython is not defined`

Maybe @SimonKrughoff knows about the distinction between `%matplotlib ipympl` and `%matplotlib widget`. Most of the documentation I find refers to `%matplotlib widget` in a JupyterLab context.

Reply:
I'm afraid I don't know much more than what you said.  Most of what I see out there says to use `%matplotlib widget`, so that's what I tend to do.  It would be great to get a better explanation of all the magics we use with vis stuff like bokeh `inline` etc.